File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter
def solution(participant, completion):
    c_par = Counter(participant)
    for n in completion:
        c_par[n] -= 1
        if c_par[n] == 0:
            c_par.pop(n)
    for i in c_par:
        answer = i
        return answer

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.util.Map.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> map = new HashMap<>();
        for(String part: participant){
            if(!map.containsKey(part))
                map.put(part, 1);
            else
                map.put(part, map.get(part) + 1);
        }
        for(String comp: completion){
            map.put(comp, map.get(comp) - 1);
        }
        for(Entry<String, Integer> entry: map.entrySet()){
            if(entry.getValue() == 1)
                return entry.getKey();
        }
        return answer;
    }
}